css/src/components/progress-bar.scss

```
@@ -0,0 +1,99 @@
+$progress-bar-height-medium: 2px;
+$progress-bar-height-large: 4px;
+
+$progress-bar-background: $secondary-background !default;
+$progress-bar-color: $primary !default;
+$progress-bar-success-color: $success !default;
+$progress-bar-warning-color: $warning !default;
+$progress-bar-danger-color: $danger !default;
+$progress-bar-border-radius-rounded: $border-radius-rounded !default;
+
+$progress-bar-animation-duration: 2s;
+
+.progress-bar-container {
+	position: relative;
+	width: 100%;
+	border-radius: $progress-bar-border-radius-rounded;
+	background-color: $progress-bar-background;
+	overflow: hidden;
+
+	.theme-high-contrast & {
+		background-color: $secondary-background-glow-high-contrast;
+	}
+
+	&.progress-bar-md {
+		height: $progress-bar-height-medium;
+	}
+
+	&.progress-bar-lg {
+		height: $progress-bar-height-large;
+	}
+
+	&.progress-bar-square {
+		border-radius: 0;
+	}
+
+	&.progress-bar-rounded {
+		border-radius: $progress-bar-border-radius-rounded;
+	}
+}
+
+@mixin indeterminate-gradient($color) {
+	position: absolute;
+	width: 100%;
+	animation: indeterminate-animation $progress-bar-animation-duration infinite linear;
+	background: linear-gradient(to right, transparent 0%, $color 50%, transparent 100%);
```

Should we add an adjustment for RTL?

```suggestion
	background: linear-gradient(to right, transparent 0%, $color 50%, transparent 100%);
	
	&:dir(rtl) {
		background: linear-gradient(to left, transparent 0%, $color 50%, transparent 100%);
	}
```